### ISSUE TITLE
SEVERE: Not in an event, but trying to add child

### ISSUE THREAD

**Issue by drockken** [triplea-game/triplea#4445]
### Engine version

1.10.13296

### My Operating System
Windows 10/64 bit

### Map name
ww2global40_2nd_edition.xml

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
I do not know how to recreate it.  Here is how it occurred:  I was beginning Germany's combat move phase.  Combat started for strategic bomb raid resolutions.  The game generated an SEVERE message (see below).  I saved the game almost immediately after the SEVERE WARNING occurred.  Zip is attached - if you load the saved game, and look at the combat history you will see a battle showing "???" Germans attacks [sic] with 2 units heading to Novgorod.

### Do you have the exact error text? Please copy/paste if so
```
Dec 20, 2018 8:42:36 PM games.strategy.engine.message.unifiedmessenger.UnifiedMessenger invoke
WARNING: Cant add an event, not a step. Must be in a step to add an event to the step. 
Trying to add event: Air Battle in Novgorod
java.lang.IllegalStateException: Cant add an event, not a step. Must be in a step to add an event to the step. 
Trying to add event: Air Battle in Novgorod
	at games.strategy.engine.history.HistoryWriter.startEvent(HistoryWriter.java:120)
	at games.strategy.engine.framework.ServerGame$1.startHistoryEvent(ServerGame.java:114)
	at games.strategy.engine.framework.ServerGame$1.startHistoryEvent(ServerGame.java:105)
	at sun.reflect.GeneratedMethodAccessor21.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at games.strategy.engine.message.unifiedmessenger.EndPoint.invokeSingle(EndPoint.java:150)
	at games.strategy.engine.message.unifiedmessenger.EndPoint.invokeMultiple(EndPoint.java:133)
	at games.strategy.engine.message.unifiedmessenger.EndPoint.invokeLocal(EndPoint.java:119)
	at games.strategy.engine.message.unifiedmessenger.UnifiedMessenger.invoke(UnifiedMessenger.java:142)
	at games.strategy.engine.message.UnifiedInvocationHandler.invoke(UnifiedInvocationHandler.java:48)
	at com.sun.proxy.$Proxy13.startHistoryEvent(Unknown Source)
	at games.strategy.engine.history.DelegateHistoryWriter.startEvent(DelegateHistoryWriter.java:29)
	at games.strategy.triplea.delegate.GameDelegateHistoryWriter.startEvent(GameDelegateHistoryWriter.java:32)
	at games.strategy.triplea.delegate.AirBattle.fight(AirBattle.java:105)
	at games.strategy.triplea.delegate.BattleTracker.fightAirRaidsAndStrategicBombing(BattleTracker.java:1129)
	at games.strategy.triplea.delegate.BattleTracker.fightAirRaidsAndStrategicBombing(BattleTracker.java:1113)
	at games.strategy.triplea.delegate.BattleDelegate.start(BattleDelegate.java:101)
	at games.strategy.engine.framework.ServerGame.startStep(ServerGame.java:512)
	at games.strategy.engine.framework.ServerGame.runStep(ServerGame.java:410)
	at games.strategy.engine.framework.ServerGame.startGame(ServerGame.java:282)
	at java.util.Optional.ifPresent(Unknown Source)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.launchInNewThread(LocalLauncher.java:41)
	at games.strategy.engine.framework.startup.launcher.AbstractLauncher.lambda$launch$0(AbstractLauncher.java:49)
	at java.lang.Thread.run(Unknown Source)

Dec 20, 2018 8:42:37 PM games.strategy.engine.history.HistoryWriter addChildToEvent
SEVERE: Not in an event, but trying to add child:Germans attacks with 2 units heading to Novgorod current is:Combat
```

### Instead of this error, what should have happened?
Strategic bombing should have been resolved without anomalous error messages

**Comment by panther2** [comment]
@drockken  Can you please add the savegame? It is missing...

**Comment by drockken** [comment]
Weird, I had added it.  Let me try again.
[cae.zip](https://github.com/triplea-game/triplea/files/2703266/cae.zip)

**Comment by drockken** [comment]
I have version 1.8.0_121.

**Comment by ron-murhammer** [comment]
@panther2 I looked over all the save games and can't reproduce the error. I re-created his combat moves and it seems to play out fine with the Hard AI as Russia. Here is the save at the end of Germany's combat move with the same moves made: 
[test.zip](https://github.com/triplea-game/triplea/files/2717403/test.zip)

I'll take a closer look at the code but the error message indicates it doesn't have anything to do with the AI. If anything there might be some sort of race condition where the air battle is trying to start before the history is ready to record it.

**Comment by ron-murhammer** [comment]
I've a improved the logging around the history writer so that if this happens again we'll have a better idea of what's causing it #4466. This error shouldn't have any negative impact on game play just that the history panel shows a missing message.

Closing this for now until we either get new info or can reproduce.
